src/script/descriptor.cpp

```diff
@@ -410,7 +411,7 @@ class BIP32PubkeyProvider final : public PubkeyProvider
         }
         return true;
     }
-    bool ToNormalizedString(const SigningProvider& arg, std::string& out, bool priv) const override
+    bool ToNormalizedString(const SigningProvider& arg, std::string& out, bool priv, std::optional<DescriptorCache> cache) const override
```

Comment:
Passing a copy of the entire cache seems overkill here, and especially with std::move's you're going to be making copies in every step in the call graph.

The extreme version would be `std::optional<std::reference_wrapper<const DescriptorCache>>` to achieve something like an optional reference to a const cache. But just `const DescriptorCache*` works just as well...

Reply:
Argh, I meant to click comment, not approve...

Reply:
Done